test/framework/README.md

```diff
@@ -0,0 +1,38 @@
+# CAPI e2e testing framework
+
+This framework aims to define common end-to-end patterns that can be reused across Cluster API providers.
+
+## Usage
+
+Use this framework as you see fit. If there are pieces that don't work for you, do not use them. If there are pieces
+that almost work for you, customize them. If this does not fit your e2e testing please file an issue and we can discuss
+your use case and find a nice solution for everyone.
+
+### Contents
+
+This framework contains
+
+* An interface to a management cluster
+    * An struct that implements the management cluster interface using `kind` as the backend.
+* A series of behavioral tests
```

Comment:
I know this was mentioned in the CAEP, but wondering now if the framework itself have tests? Maybe we should define test suites that contain the actual tests and the framework should just have setup/teardown/validation helpers?

Reply:
I see what you're saying. I think the term "framework" may be a bit ambiguous. I'm using framework/library here interchangeably. This framework defines tests that providers may or may not use. I think it would also be a nice add-on to provide some kind of suite that wraps up tests. Part of the framework/library is to provide functions for setup and teardown. Take a  look at this PR to see how this is designed to be used: https://github.com/kubernetes-sigs/cluster-api/pull/1742